js/core/devices.js

```diff
@@ -13,7 +12,6 @@ import { sessionStorage as SessionStorage } from './utils/storage';
 import { changeCallback, value as viewPort } from './utils/view_port';
 import Config from './config';
 
-const navigator = getNavigator();
```

Comment:
afaik, navigator is a global object like window. Why would you move it to local variables, instead of keeping it here like _window_ var below?

Reply:
I change it because I need to mock the navigator in QUnit tests.
The [getNavigator](https://github.com/DevExpress/DevExtreme/blob/23_1/js/core/utils/window.js#L48) method returns the navigator from the custom window object from a closure.
And we can set a custom window object (mock) with the [setWindow](https://github.com/DevExpress/DevExtreme/blob/23_1/js/core/utils/window.js#L18) method.

So, if I leave this as is

I should mock the window before I import devices.js in QUnit's test file.
(I'm not 100% sure that it will work, but looks like it will)
From my point of view, it looks unobvious and ugly, so I decided to get navigator from the getNavigator method in functions runtime instead of declaring a variable in this scope :)